File: tests/fixtures/typescript/code_duplication.ts
function processUserReport(users: any[]): any[] {
    const report: any[] = [];
    for (const user of users) {
        const entry: any = {};
        entry.id = user.id;
        entry.name = user.name;
        entry.email = user.email;
        entry.status = user.active ? "active" : "inactive";
        entry.formattedDate = user.createdAt;
        entry.displayName = `${user.name} (${user.email})`;
        report.push(entry);
    }
    return report;
}

function processAdminReport(admins: any[]): any[] {
    const report: any[] = [];
    for (const admin of admins) {
        const entry: any = {};
        entry.id = admin.id;
        entry.name = admin.name;
        entry.email = admin.email;
        entry.status = admin.active ? "active" : "inactive";
        entry.formattedDate = admin.createdAt;
        entry.displayName = `${admin.name} (${admin.email})`;
        report.push(entry);
    }
    return report;
}

function processVendorReport(vendors: any[]): any[] {
    const report: any[] = [];
    for (const vendor of vendors) {
        const entry: any = {};
        entry.id = vendor.id;
        entry.name = vendor.name;
        entry.email = vendor.email;
        entry.status = vendor.active ? "active" : "inactive";
        entry.formattedDate = vendor.createdAt;
        entry.displayName = `${vendor.name} (${vendor.email})`;
        report.push(entry);
    }
    return report;
}

File: tests/fixtures/python/code_duplication.py
"""Functions with duplicated logic (future: duplication detection)."""


def process_user_report(users):
    report = []
    for user in users:
        entry = {}
        entry["id"] = user.id
        entry["name"] = user.name
        entry["email"] = user.email
        entry["status"] = "active" if user.active else "inactive"
        entry["formatted_date"] = user.created_at.strftime("%Y-%m-%d")
        entry["display_name"] = f"{user.name} ({user.email})"
        report.append(entry)
    return report


def process_admin_report(admins):
    report = []
    for admin in admins:
        entry = {}
        entry["id"] = admin.id
        entry["name"] = admin.name
        entry["email"] = admin.email
        entry["status"] = "active" if admin.active else "inactive"
        entry["formatted_date"] = admin.created_at.strftime("%Y-%m-%d")
        entry["display_name"] = f"{admin.name} ({admin.email})"
        report.append(entry)
    return report


def process_vendor_report(vendors):
    report = []
    for vendor in vendors:
        entry = {}
        entry["id"] = vendor.id
        entry["name"] = vendor.name
        entry["email"] = vendor.email
        entry["status"] = "active" if vendor.active else "inactive"
        entry["formatted_date"] = vendor.created_at.strftime("%Y-%m-%d")
        entry["display_name"] = f"{vendor.name} ({vendor.email})"
        report.append(entry)
    return report
